Client/Frontend/Toolbar+URLBar/TabLocationView.swift

```diff
@@ -45,10 +44,18 @@ class TabLocationView: UIView {
     var url: URL? {
         didSet {
             updateTextWithURL()
-            trackingProtectionButton.isHidden = isTrackingProtectionHidden
+            trackingProtectionButton.isHidden = isValidHttpUrlProtocol
+            shareButton.isHidden = !shouldEnableShareButtonFeature && isValidHttpUrlProtocol
             setNeedsUpdateConstraints()
         }
     }
+    
+    var shouldEnableShareButtonFeature: Bool {
+        guard featureFlags.isFeatureEnabled(.shareSheetChanges, checking: .buildOnly) else {
```

Comment:
We should use `shareToolbarChanges`. I added to separate flags to have flexibility in case we want to experiment the ShareSheet changes separately from the toolbar changes